core/src/main/java/jenkins/JavaVMArguments.java

```diff
@@ -0,0 +1,45 @@
+package jenkins;
+
+import hudson.Functions;
+import hudson.util.ProcessTree;
+import hudson.util.jna.GNUCLibrary;
+import java.lang.management.ManagementFactory;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+/**
+ * List of arguments for Java VM and application.
+ */
+@Restricted(NoExternalUse.class)
+public class JavaVMArguments {
```

Comment:
Compare with [the original](https://github.com/kohsuke/akuma/blob/master/src/main/java/com/sun/akuma/JavaVMArguments.java), a 546-line hornet's nest that largely duplicated the logic in core's `ProcessTree` class.